moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/ompl_constraints.h

```diff
@@ -0,0 +1,325 @@
+
+  const std::string& getLinkName()
+  {
+    return link_name_;
+  }
+
+  const Eigen::Vector3d getTargetPosition()
+  {
+    return target_position_;
+  }
+
+  const Eigen::Quaterniond getTargetOrientation()
+  {
+    return target_orientation_;
+  }
+
+protected:
+  /** \brief Thread-safe storage of the robot state.
+   *
+   * The robot state is modified for kinematic calculations. As an instance of this class is possibly used in multiple
+   * threads due to OMPL's LazeGoalSampler, we need a separate robot state in every thread.
+   * */
+  TSStateStorage state_storage_;
+  const robot_state::JointModelGroup* joint_model_group_;
+
+  // all attributes below can be considered const as soon as the constraint message is parsed
+  // but I (jeroendm) do not know how to elegantly express this in C++
+  // parsing the constraints message and passing all this data members separately to the constructor
+  // is a solution, but it adds complexity outside this class, which is also not ideal.
+
+  /** \brief Robot link the constraints are applied to. */
+  std::string link_name_;
+
+  /** \brief Upper and lower bounds on constrained variables. */
+  std::vector<Bounds> bounds_;
+
+  /** \brief target for equality constraints, nominal value for inequality constraints. */
+  Eigen::Vector3d target_position_;
+
+  /** \brief target for equality constraints, nominal value for inequality constraints. */
+  Eigen::Quaterniond target_orientation_;
+
+public:
+  // Macro for classes containing fixed size eigen vectors that are dynamically allocated when used.
+  // https://eigen.tuxfamily.org/dox/group__TopicStructHavingEigenMembers.html
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+/******************************************
+ * Box constraint
+ * ****************************************/
+/** \brief Box shaped position constraints
+ *
+ * Reads bounds on x, y and z position from a position constraint
+ * at constraint_region.primitives[0].dimensions.
+ * Where the primitive has to be of type `shape_msgs/SolidPrimitive.BOX`.
+ *
+ * These bounds are applied around the nominal position and orientation
+ * of the box.
+ * */
+class BoxConstraint : public BaseConstraint
+{
+public:
+  BoxConstraint(const robot_model::RobotModelConstPtr& robot_model, const std::string& group,
+                const unsigned int num_dofs);
+  virtual void parseConstraintMsg(const moveit_msgs::Constraints& constraints) override;
+  virtual Eigen::VectorXd calcError(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+  virtual Eigen::MatrixXd calcErrorJacobian(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+};
+
+/******************************************
+ * Equality Position Constraints
+ * ****************************************/
+/** \brief Equality constraints on a link's position.
+ *
+ *  When you set the name of a constraint to 'use_equality_constraints', all constraints with a dimension lower that
```

Comment:
I used the "name" field in [moveit_msgs/Constraints](http://docs.ros.org/latest/api/moveit_msgs/html/msg/Constraints.html), but it is hacky.

**I need to remove this comment** as it will not be necessary anymore if I add change the orientation constraint message as discussed in [this issue](https://github.com/ros-planning/moveit_msgs/issues/89). I started working on it but my pace is not the same as it was this summer...